### PR TITLE
Viewer: add ability to load and save property values from/to json

### PR DESCRIPTION
**Pull request by ogoffart**
issue #238

### CODE REVIEW

File: tools/viewer/main.rs [hunk 1]
```diff
@@ -39,13 +39,26 @@ struct Cli {
...
+    save_data: Option<std::path::PathBuf>,
 }
 
 thread_local! {static CURRENT_INSTANCE: std::cell::RefCell<Option<ComponentInstance>> = Default::default();}
 
 fn main() -> Result<()> {
     let args = Cli::from_args();
 
+    if args.auto_reload && (args.save_data.is_some()) {
+        eprintln!("Cannot pass both --auto-reload and --save-data");
```
Comment (tronical) [inline tools/viewer/main.rs:58]: Why not? :)

File: tools/viewer/main.rs [hunk 2]
```diff
@@ -39,13 +39,26 @@ struct Cli {
...
+    #[structopt(long, name = "save data file")]
+    save_data: Option<std::path::PathBuf>,
 }
 
 thread_local! {static CURRENT_INSTANCE: std::cell::RefCell<Option<ComponentInstance>> = Default::default();}
 
 fn main() -> Result<()> {
     let args = Cli::from_args();
 
+    if args.auto_reload && (args.save_data.is_some()) {
```
Comment (tronical) [inline tools/viewer/main.rs:57]:
```suggestion
    if args.auto_reload && args.save_data.is_some() {
```

**Review by tronical — APPROVED** [review]

File: tools/viewer/main.rs (hunk 1 above)
Comment (ogoffart) [reply tools/viewer/main.rs:58]: When do we save the properties?

Comment (tronical) [reply tools/viewer/main.rs:58]: Before reloading, and then maybe load again with the new ui loaded?

Kind of a way to persist properties across ui edit cycles.

Comment (tronical) [reply tools/viewer/main.rs:58]: (Btw, just a side comment - don’t let that hold you off from merging :)

**Pull request merged**